standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -1896,7 +1917,12 @@ private MTable getMTable(String catName, String db, String table) {
           }
         } else if (projectionFields.size() == 1) {
           // Execute the query to fetch the partial results.
-          List<Object> results = (List<Object>) query.execute(db, catName, lowered_tbl_names);
+          List<Object[]> results = new ArrayList<>();
+          if(tablePattern == null) {
```

Comment:
Shouldn't there be 3 possiblities here?
tablePattern != null && !lowered_tbl_names.isEmpty() --> execute() needs 4 arguments
lowered_tbl_names.isEmpty() && tablePattern != null --> execute() needs 3 arguments
!lowered_tbl_names.isEmpty() && tablePattern == null --> execute() needs 3 arguments